File: P250137.py
def solution(bandage, health, attacks):
    answer = 0
    time = 0
    attack_time = [0] * 1001
    last = 0
    for attack in attacks:
        attack_time[attack[0]] = attack[1]
        if attack[0] > last:
            last = attack[0]
    
    hp = health    
    continous = 0
    while True:
        time += 1
        
        if time > last:
            answer = hp
            break
        
        if attack_time[time]:
            continous = 0
            hp = max(hp - attack_time[time], 0)
        
        if not hp:
            answer = -1
            break
        
        if not attack_time[time]:
            hp += bandage[1]
            continous += 1
            if continous == bandage[0]:
                hp += bandage[2]
                continous = 0
            
            hp = min(hp, health)
        
    return answer

File: P250137.java
class P250137 {
    public int solution(int[] bandage, int health, int[][] attacks) {
        int answer = 0;
        int time = 0;
        int[] attackTime = new int[1001];
        int last = 0;
        
        for (int[] attack : attacks) {
            attackTime[attack[0]] = attack[1];
            if (attack[0] > last) last = attack[0];
        }
        
        int hp = health;
        int continous = 0;
        
        while (true) {
            time += 1;
            
            if (time > last) {
                answer = hp;
                break;
            }
            
            if (attackTime[time] > 0) {
                continous = 0;
                hp = hp - attackTime[time] > 0 ? hp-attackTime[time] : 0;
            }
            
            if (hp == 0) {
                answer = -1;
                break;
            }
            
            if (attackTime[time] == 0) {
                hp += bandage[1];
                continous += 1;
                
                if (continous == bandage[0]) {
                    hp += bandage[2];
                    continous = 0;
                }
                
                hp = hp < health ? hp : health;
            }
        }
        return answer;
    }
}